### DISCUSSION TITLE
goroutine pool对性能优化的疑问

### DISCUSSION THREAD

**Discussion by hugh-404** [panjf2000/ants#224] [Q&A] [answered]
在go GMP模型中，数据结构G本身会被go scheduler复用，从内存和性能方面考虑，是不是和goroutine pool中对goroutine的复用效果是一样的呢

**Comment by panjf2000** [comment, marked as answer]
不只是复用，goroutine pool 还有一个主要的特性是限制，go原生的复用不会做数量的限制。还有就是 pool 其实是 worker 模式，相当于提前起动了一批 goroutines 在跑，不用每次都重新创建。

**Reply by Minnull** [reply]
请问一下，如果提交到pool的全部是**长期运行的task**。使用ants和原生goroutine的对比，内存等方面有明显优势吗？

**Reply by panjf2000** [reply]
这种情况就没啥优势了，如果数量也不是很多的话，建议直接用原生的 `go` 关键字就行了。

**Comment by Tochemey** [comment]
@panjf2000 Please for those of us who do not understand the mandarin, it will be difficult for us to know whether an issue we are facing has been resolved in those discussions. It will be nice if we can have some context as well :)

**Reply by panjf2000** [reply]
I am sorry for this confusion about the language. If it were up to me, I'd prefer to use English in all GitHub issues, pull requests, and discussions, after all, this is an open-source project for the whole world. That being said, I can only advocate for people to use English while engaging in my open-source projects, which is also what I've been doing all the time, but I can't force them to, literally, I can't.

Plenty of Chinese users use my open-source projects, and they might not be very fluent in English, so they prefer to use Chinese knowing that I can also read Chinese.

**Reply by panjf2000** [reply]
As for this discussion, it's just some general questions about the working principle of `ants`, not issues. So don't worry about it.
